### PR TITLE
Issue #84 Authentication chaining does not work if the user search attribute of the LDAP authentication module does not match that of the data store

### PR DESCRIPTION
**Pull request by kurotsu**
## Analysis

See, #84 "Description".

## Solution

The following property added to the ldap authentication module in the OpenAM console.
 "Overwrite User Name in sharedState upon Authentication Success"
When this property is enabled,  The uid (`Attribute Used to Retrieve User Profile`) obtained from LDAP is stored in sharedState. Then subsequent authentications can handle DataStore successfully.
 
## Testing

* Try #84 "Steps to reproduce".
   At that time, in the OpenAM console Enable the following property added to the ldap authentication module.
   "Overwrite User Name in sharedState upon Authentication Success"
 Then, Try to enable or disable the following property added to the ldap authentication module.
   "Return User DN to DataStore"

## Regression testing

* Try #84 "Steps to reproduce"

### CODE REVIEW

File: openam-authentication/openam-auth-ldap/src/main/resources/amAuthLDAP.properties [hunk 1]
```diff
@@ -26,6 +26,7 @@
 #
 # Portions Copyrighted 2011-2016 ForgeRock AS.
 # Portions Copyrighted 2012 Open Source Solution Technology Corporation
+# Portions Copyrighted 2019 Open Source Solution Technology Corporation
```
Comment (tsujiguchitky) [inline openam-authentication/openam-auth-ldap/src/main/resources/amAuthLDAP.properties]: `2019` -> `2012-2019`. 
And please delete previous line.

File: openam-authentication/openam-auth-ldap/src/main/resources/amAuthLDAP_ja.properties [hunk 2]
```diff
@@ -28,6 +28,7 @@
 # Portions Copyrighted 2012-2015 ForgeRock AS.
 # Portions Copyrighted 2014 Open Source Solution Technology Corporation
 # Portions Copyrighted 2014 Nomura Research Institute, Ltd
+# Portions Copyrighted 2019 Open Source Solution Technology Corporation
```
Comment (tsujiguchitky) [inline openam-authentication/openam-auth-ldap/src/main/resources/amAuthLDAP_ja.properties]: `2019` -> `2014-2019`.
And please delete previous line.

**Comment by tsujiguchitky** [comment]
It seems that `Return User DN to DataStore` needs to be disabled in order to work as expected.  I think that it is good not to depend on this setting.

File: openam-authentication/openam-auth-ldap/src/main/resources/amAuthLDAP.properties (hunk 1 above)
Comment (kurotsu) [reply openam-authentication/openam-auth-ldap/src/main/resources/amAuthLDAP.properties]: IC.

File: openam-authentication/openam-auth-ldap/src/main/resources/amAuthLDAP_ja.properties (hunk 2 above)
Comment (kurotsu) [reply openam-authentication/openam-auth-ldap/src/main/resources/amAuthLDAP_ja.properties]: IC.

**Comment by kurotsu** [comment]
I dealt with reviews.

**Comment by kurotsu** [comment]
The following properties are no longer dependent on enabled or disabled.
"Return User DN to DataStore"

**Comment by tsujiguchitky** [comment]
> The following properties are no longer dependent on enabled or disabled.
> "Return User DN to DataStore"

OK, Thanks!

File: openam-core/src/main/java/org/forgerock/openam/ldap/LDAPAuthUtils.java [hunk 3]
```diff
@@ -1320,6 +1320,13 @@ public String getUserId(String name) {
         }
     }
 
+    /**
+     * TODO-JAVADOC
```
Comment (tsujiguchitky) [inline openam-core/src/main/java/org/forgerock/openam/ldap/LDAPAuthUtils.java]: Don't omit comments...
And please add a copyright header to this file.

Comment (kurotsu) [reply openam-core/src/main/java/org/forgerock/openam/ldap/LDAPAuthUtils.java]: IC.

**Comment by kurotsu** [comment]
Done.

**Comment by tsujiguchitky** [comment]
LGTM

**Pull request merged**